ambry-store/src/main/java/com/github/ambry/store/StorageManager.java

```diff
@@ -536,6 +536,29 @@ DiskManager addDisk(DiskId diskId) {
     });
   }
 
+  /**
+   * Add a new store to the storage manager for file copy based replication post filecopy is completed.
+   * @param replica the {@link ReplicaId} of the {@link Store} which would be added.
+   * @return
+   */
+  @Override
+  public boolean addBlobStoreForFileCopy(ReplicaId replica) {
+    if (!partitionToDiskManager.containsKey(replica.getPartitionId())) {
+      logger.info("PartitionId {} doesn't exist in storage manager during state build, rejecting adding store request", replica.getPartitionId());
+      return false;
+    }
+    // We don't require addDisk since DiskManager is already started during initialization of StorageManager as part
+    // of prefilecopy steps. We will fetch it from partitionToDiskManager map.
+    DiskManager diskManager = partitionToDiskManager.get(replica.getPartitionId());
+    if (diskManager == null || !diskManager.addBlobStoreForFileCopy(replica)) {
+      logger.error("Failed to add new store into DiskManager");
+      return false;
+    }
+    partitionNameToReplicaId.put(replica.getPartitionId().toPathString(), replica);
```

Comment:
Same as this: https://github.com/linkedin/ambry/pull/2954/files#r1894565174

Reply:
Done, we should handle it in prefilecopy. Please make a note.